Assets/Scripts/AdsAndIAP/IAPController.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Authors: Timothy Ng
+ * Description: Handles purchases
+ * Usage: [no notes]
+ */
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Purchasing;
+
+public class IAPController : SingletonController<IAPController>, IStoreListener
+{
+    const string PRODUCT_1000_GOLD = "gold1000";
+
+    IStoreController m_StoreController;
+    IExtensionProvider m_StoreExtensionProvider;
+
+    #region Static Accessors
+
+    // Returns the Instance cast to the sublcass
+    public static IAPController GetInstance
+    {
+        get
+        {
+            return Instance as IAPController;
+        }
+    }
+
+    #endregion
+
+    #region MonoBehaviourExtended Overrides
+    protected override void fetchReferences()
+    {
+        base.fetchReferences();
+        InitializePurchasing();
+    }
+    #endregion
+
+    #region IStoreListener Overrides
```

Comment:
Newline after #region

Also the convention when it's an interface is to write "IStoreListener Interface"